vhost-device-i2c/src/main.rs

```diff
@@ -310,8 +332,7 @@ mod tests {
     fn test_parse_successful() {
         let socket_name = "vi2c.sock";
 
-        // Space before and after the device list and socket name
-        let cmd_args = I2cArgs::from_args(" ./vi2c.sock", " 1:4 ", 1);
+        let cmd_args = I2cArgs::from_args("./vi2c.sock", "1:4", 1);
```

Comment:
Why isn't this required anymore ?

Reply:
This was discussed here before: https://github.com/rust-vmm/vhost-device/pull/446#discussion_r1327001830. Space may be a fine part of a filename :)

Reply:
Space between the characters should be fine, but trim was only removing the trailing and leading whitespaces.. I guess with pathbuf now, we will error out in such cases where an extra space is provided before the file name ?

Reply:
@vireshk The same happens with `cat`:

```
❯ cat ' tmp'
cat: ' tmp': No such file or directory
```
So I do not thinkg we should trim spaces if they can be a valid part of a filename. This only happens if you go out of your way to quote the string. So the user just passed us either a valid filename with a space or we error out when we try to create the socket because the path is invalid. Trimming will just silently use a different path than what the user specified.

Reply:
Yeah, I was just asking if we will error out. That's fine. Maybe we should keep the tests around then, and look for failure instead of success.

Reply:
Not sure if there's a point to be honest... if you keep the spaces, it will look for a "` .`" directory and a `vi2c.sock` file under it and error out because ` .` does not exist. When you test for this:

```
let cmd_args = I2cArgs::from_args(" ./vi2c.sock", " 1:4", 1);
```

It's like running this shell command:

```shell
vhost-device-i2c -s " ./vi2c.sock" --device-list " 1:4" --socket-count 1
```

Which is not the same as 

```shell
vhost-device-i2c -s   ./vi2c.sock --device-list   1:4 --socket-count 1
```

Here, the shell trims the whitespace.

Reply:
Right, but it is okay for someone to put the arguments under double quotes, specially the device list as it can get long and tricky like: 1:4:5,2:2:6,3:7:9..

And so making sure that we fail properly if a space is added there by mistake can be useful ?

Reply:
@vireshk the device list part still trims the string, I didn't change that. Do we want to disallow it? WDYT?

Reply:
No.. Looks like there is some confusion here, lets start again.

I think it is fine for anyone to pass both the arguments, socket-path and device list within double quotes. 

Disallowing path with a leading or trailing space is fine, I am suggesting we add a failure test for same to make sure we fail appropriately.

There is no need to disallow the device list string for that, and so the earlier test which had a leading space should be kept intact to make sure it works fine.

Reply:
@vireshk ah that's how I thought about it as well. I pushed some changes. I added a commit for path validation and re-added the device list trimming test.